mattermost-notify/action.yml

```diff
@@ -8,49 +9,61 @@ inputs:
+  commit-message:
+    description: Git commit message to use
+  repository:
+    description: GitHub repository (org/repo). Default is github.repository
+    default: ${{ github.repository }}
+  workflow:
+    description: GitHub workflow ID
+  workflow-name:
+    description: GitHub workflow name
+
 branding:
   icon: "package"
   color: "green"
 
 runs:
   using: "composite"
   steps:
-    - name: Set up Python
-      uses: actions/setup-python@v4
-      id: python
+    - name: Install
+      uses: greenbone/actions/pipx@v3
+      id: pipx
       with:
         python-version: "3.10"
-    - name: Virtual Environment
-      id: virtualenv
-      run: |
-        echo "path=${{ github.action_path }}/mattermost-notify-venv" >> $GITHUB_OUTPUT
-        echo "name=mattermost-notify-venv" >> $GITHUB_OUTPUT
-      shell: bash
-    - name: Cache Virtual Environment
-      id: cache-virtualenv
-      uses: actions/cache@v3
-      with:
-        key: ${{ runner.os }}-${{ steps.python.outputs.python-version }}-${{ steps.virtualenv.outputs.name }}
-        path: ${{ steps.virtualenv.outputs.path }}
-    - name: Create virtual environment
-      if: ${{ steps.cache-virtualenv.outputs.cache-hit != 'true' }}
-      run: |
-        python -m venv ${{ steps.virtualenv.outputs.path }}
-      shell: bash
-    - run: |
-        source ${{ steps.virtualenv.outputs.path }}/bin/activate
-        python -m pip install --upgrade pip
-        python -m pip install --upgrade mattermost-notify
-      shell: bash
+        install: mattermost-notify
     - name: Create the Mattermost Message
       shell: bash
       run: |
-        source ${{ steps.virtualenv.outputs.path }}/bin/activate
-        if [ "${{ inputs.MATTERMOST_HIGHLIGHT }}" ] ; then
-          mnotify-git ${{ inputs.MATTERMOST_WEBHOOK_URL }} ${{ inputs.MATTERMOST_CHANNEL }} \
-            --highlight ${{ inputs.MATTERMOST_HIGHLIGHT }}
-        else
-          mnotify-git ${{ inputs.MATTERMOST_WEBHOOK_URL }} ${{ inputs.MATTERMOST_CHANNEL }}
+        ARGS=""
+        if [ -n "${{ inputs.commit }}" ]; then
+          ARGS="${ARGS} --commit ${{ inputs.commit }}"
         fi
+        if [ -n "${{ inputs.commit-message }}" ]; then
+          ARGS="${ARGS} --commit-message ${{ inputs.commit-message }}"
+        fi
+        if [ -n "${{ inputs.branch }}" ]; then
+          ARGS="${ARGS} --branch ${{ inputs.branch }}"
+        fi
+        if [ -n "${{ inputs.repository }}" ]; then
+          ARGS="${ARGS} --repository ${{ inputs.repository }}"
+        fi
+        if [ -n "${{ inputs.workflow }}" ]; then
+          ARGS="${ARGS} --workflow ${{ inputs.workflow }}"
+        fi
+        if [ -n "${{ inputs.workflow-name }}" ]; then
+          ARGS="${ARGS} --workflow_name ${{ inputs.workflow-name }}"
+        fi
+        if [ -n "${{ inputs.MATTERMOST_HIGHLIGHT }}" -o -n "${{ inputs.highlight }}" ] ; then
+          ARGS="--highlight ${{ inputs.MATTERMOST_HIGHLIGHT }} ${{ inputs.highlight }}"
+        fi
```

Comment:
Urgh ... wish this could be done more "easily". :/

Reply:
Yes me too but that's how it works best. At least at the moment.